src/shared/components/MapLayout.js

```diff
@@ -1,33 +1,63 @@
 import React, {PropTypes, Component} from 'react';
 import { connect } from 'react-redux';
 
-import { getMapMarkers } from '../actions/map';
+import { getMapMarkers, selectMapMarker, addMapMarker } from '../actions/map';
 import MapBlock from './MapBlock.js';
+import MapCard from './MapCard.js';
 
 class MapLayout extends Component {
 
+  constructor(props) {
+    super(props);
+    this.handleSelectMarker = this.handleSelectMarker.bind(this);
+  }
+
   componentDidMount() {
     this.props.getMapMarkers().then(
       () => {
```

Comment:
Extract properties rather than access to props with many dots:

const {
  getMapMarkers,
  addMapMaker
} = this.props;